spec/function.dd

```diff
@@ -1807,25 +1808,23 @@ ref int escape()
 ---
 )
         $(P The `return` attribute can also be used for wrapper structs to limit
-        the lifetime of the returned value:
+        the lifetime of the returned value, even when the method is not `ref`:
         )
 
-$(SPEC_RUNNABLE_EXAMPLE_COMPILE
+$(SPEC_RUNNABLE_EXAMPLE_FAIL
 ---
-@safe:
-
 struct S
 {
     private int* p;
-    int* get() return => p;
+    int* get() return @safe => p;
 }
 
-void f()
+void f() @safe
 {
     int* p;
     {
         S s;
-        //p = s.get(); // deprecated
+        p = s.get(); // error, p has longer lifetime than s
         p = (new S).get(); // OK, heap allocated S
     }
```

Comment:
This can be compacted and be made more expressive without the `{}` scope:
```d
struct S
{
    private int p;
    int* get() return @safe => &p;
}
void main() @safe
{
    int* p = S().get(); // error, `p` has longer lifetime than temporary S()
    int* q = (new S).get(); // OK, heap allocated
}
```
First, returning the address of the `int` field – opposed to the copy of a pointer-type field – makes it obvious why there is a `return` attribute.
Second `S().get()` removes the necessity for the scope.

Reply:
I have extended the example to make the scope more useful.